### PR TITLE
Refactor pkg/kubelet/kubelet.go: shouldContainerBeRestarted().

### PR DESCRIPTION
**Pull request by yifan-gu**
Simplify the function signature.

Just minor changes to make the function's input 'docker unrelated'.
PTAL, thank you! @vmarmol

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1074,11 +1074,12 @@ func (kl *Kubelet) makePodDataDirs(pod *api.BoundPod) error {
...
 
-func (kl *Kubelet) shouldContainerBeRestarted(pod *api.BoundPod, containerName, dockerContainerName string, uid types.UID) bool {
+func (kl *Kubelet) shouldContainerBeRestarted(container *api.Container, pod *api.BoundPod) bool {
+	podFullName := GetPodFullName(pod)
 	// Check RestartPolicy for dead container
-	recentContainers, err := dockertools.GetRecentDockerContainersWithNameAndUUID(kl.dockerClient, GetPodFullName(pod), uid, containerName)
+	recentContainers, err := dockertools.GetRecentDockerContainersWithNameAndUUID(kl.dockerClient, podFullName, pod.UID, container.Name)
 	if err != nil {
-		glog.Errorf("Error listing recent containers:%s", dockerContainerName)
+		glog.Errorf("Error listing recent containers: %v", err)
```
Comment (vmarmol) [inline pkg/kubelet/kubelet.go]: maybe: Error listing recent containers for pod <podFullName>: <err>

**Comment by dchen1107** [comment]
LGTM

File: pkg/kubelet/kubelet.go [hunk 2]
```diff
@@ -1087,6 +1088,7 @@ func (kl *Kubelet) shouldContainerBeRestarted(pod *api.BoundPod, containerName,
 	}
 
 	if len(recentContainers) > 0 {
+		dockerContainerName := dockertools.BuildDockerName(pod.UID, podFullName, container)
 		if pod.Spec.RestartPolicy.Never != nil {
 			glog.Infof("Already ran container with name %s, do nothing",
```
Comment (vmarmol) [inline pkg/kubelet/kubelet.go]: I would say to just use the podFullName instead. The Docker container name is not building us anything and confusing the code.

**Comment by vmarmol** [comment]
Minor comments, looks good overall :)

File: pkg/kubelet/kubelet.go (hunk 2 above)
Comment (yifan-gu) [inline pkg/kubelet/kubelet.go]: Agree.

**Comment by vmarmol** [comment]
LGTM, thanks @yifan-gu! Will merge on green.

**Pull request merged**